### PR TITLE
Couple fixes to fi_pingpong

### PR DESCRIPTION
**Pull request by chien-intel**
First commit is cleanup.  Second commit adds source address parameter to fix issue #9900, using fi_pingpong in multi-adapter environment.

### CODE REVIEW

**Review by j-xiong — APPROVED** [review]

**Comment by zachdworkin** [comment]
bot:aws:retest

**Comment by shijin-aws** [comment]
@zachdworkin AWS CI failed due to fi_pingpong failed on Windows platform

```
15:13:10 [debug] C:\9971_1712094667\libfabric\util\pingpong.c:283 :  * PingPong options:
15:13:10 [debug] C:\9971_1712094667\libfabric\util\pingpong.c:285 :   - src_addr            : [None]
15:13:10 [debug] C:\9971_1712094667\libfabric\util\pingpong.c:286 :   - src_port            : [0]
15:13:10 [debug] C:\9971_1712094667\libfabric\util\pingpong.c:287 :   - dst_addr            : [None]
15:13:10 [debug] C:\9971_1712094667\libfabric\util\pingpong.c:288 :   - dst_port            : [0]
15:13:10 [debug] C:\9971_1712094667\libfabric\util\pingpong.c:289 :   - sizes_enabled       : default size = 64
15:13:10 [debug] C:\9971_1712094667\libfabric\util\pingpong.c:290 :   - iterations          : default iterations: 10
15:13:10 [debug] C:\9971_1712094667\libfabric\util\pingpong.c:293 :   - provider            : efa
15:13:10 [debug] C:\9971_1712094667\libfabric\util\pingpong.c:2237: Selected endpoint: RDM
15:13:10 [debug] C:\9971_1712094667\libfabric\util\pingpong.c:509 : Initializing control messages
15:13:10 socket(): C:\9971_1712094667\libfabric\util\pingpong.c:430 , ret=-22 (Invalid argument)
15:13:10 [debug] C:\9971_1712094667\libfabric\util\pingpong.c:1876: Freeing resources of test suite
15:13:10 [debug] C:\9971_1712094667\libfabric\util\pingpong.c:1910: Resources of test suite freed
15:13:10 [debug] C:\9971_1712094667\libfabric\util\pingpong.c:283 :  * PingPong options:
15:13:10 [debug] C:\9971_1712094667\libfabric\util\pingpong.c:285 :   - src_addr            : [None]
15:13:10 [debug] C:\9971_1712094667\libfabric\util\pingpong.c:286 :   - src_port            : [0]
15:13:10 [debug] C:\9971_1712094667\libfabric\util\pingpong.c:287 :   - dst_addr            : [127.0.0.1]
15:13:10 [debug] C:\9971_1712094667\libfabric\util\pingpong.c:288 :   - dst_port            : [0]
15:13:10 [debug] C:\9971_1712094667\libfabric\util\pingpong.c:289 :   - sizes_enabled       : default size = 64
15:13:10 [debug] C:\9971_1712094667\libfabric\util\pingpong.c:290 :   - iterations          : default iterations: 10
15:13:10 [debug] C:\9971_1712094667\libfabric\util\pingpong.c:293 :   - provider            : efa
15:13:10 [debug] C:\9971_1712094667\libfabric\util\pingpong.c:2237: Selected endpoint: RDM
15:13:10 [debug] C:\9971_1712094667\libfabric\util\pingpong.c:509 : Initializing control messages
15:13:10 [error] C:\9971_1712094667\libfabric\util\pingpong.c:321 : getaddrinfo : E
15:13:10 [debug] C:\9971_1712094667\libfabric\util\pingpong.c:1876: Freeing resources of test suite
15:13:10 [debug] C:\9971_1712094667\libfabric\util\pingpong.c:1910: Resources of test suite freed
15:13:10 2024-04-02 22:11:37,190 - INFO - test_runner_windows - Server process failed with return code 22
15:13:10 
15:13:10 Build step 'Execute shell' marked build as failure
15:13:10 Recording test results
```

**Comment by shijin-aws** [comment]
Did you your change modify the default behavior for OOB exchange?

**Comment by zachdworkin** [comment]
@shijin-aws #9963 did modify the default behaviour for fabtests. The backlog went from 0 to 511. This prevents SYN Flooding on port 3000 when using oob exchange in fabtests. I did not change the behaviour outside of fabtests.

File: util/pingpong.c [hunk 1]
```diff
@@ -2296,9 +2324,7 @@ int main(int argc, char **argv)
 	ct.hints->mode = FI_CONTEXT | FI_CONTEXT2 | FI_MSG_PREFIX;
 	ct.hints->domain_attr->mr_mode = FI_MR_LOCAL | OFI_MR_BASIC_MAP;
 
-	ofi_osd_init();
```
Comment (shijin-aws) [inline util/pingpong.c:2299]: why removing it?

**Comment by shijin-aws** [comment]
Is `fi_ini` necessarily called on Windows? Saw the commit that `ofi_osd_init()` is removed from the test because it is called in `fi_ini`, but I find it has pthread calls which may not be compiled on windows at all

**Comment by j-xiong** [comment]
`fi_ini` is always called from `fi_getinfo` and `fi_fabric`. There are stub definitions for pthread under `include/windows`.

**Comment by shijin-aws** [comment]
Any ideas how can this change cause `15:13:10 socket(): C:\9971_1712094667\libfabric\util\pingpong.c:430 , ret=-22 (Invalid argument)` ?

**Comment by chien-intel** [comment]
> Any ideas how can this change cause `15:13:10 socket(): C:\9971_1712094667\libfabric\util\pingpong.c:430 , ret=-22 (Invalid argument)` ?

It might be an ordering issue.  ofi_socket is called before any calls to libfabric.  I don't care that much about removing ofi_osd_init() call.  I will drop that commit from this PR.

**Comment by shijin-aws** [comment]
> > Any ideas how can this change cause `15:13:10 socket(): C:\9971_1712094667\libfabric\util\pingpong.c:430 , ret=-22 (Invalid argument)` ?
> 
> It might be an ordering issue. ofi_socket is called before any calls to libfabric. I don't care that much about removing ofi_osd_init() call. I will drop that commit from this PR.

Ah, that makes sense. Thank you!

**Comment by shijin-aws** [comment]
@chien-intel windows test passed for the latest push

**Comment by chien-intel** [comment]
Great, thanks!

**Pull request merged**